parsing/unit_info.ml

```diff
@@ -32,36 +35,38 @@ let basename_chop_extensions basename  =
   | dot_pos -> String.sub basename 0 dot_pos
   | exception Not_found -> basename
 
-let modulize s = String.capitalize_ascii s
+let strict_modulize s =
+  match Misc.UIdent.capitalize s with
+  | Ok x -> x
+  | Error _ -> raise (Error (Invalid_encoding s))
 
-(* We re-export the [Misc] definition *)
-let normalize = Misc.normalized_unit_filename
+let modulize s = match Misc.UIdent.capitalize s with Ok x | Error x -> x
 
-let modname_from_source source_file =
-  source_file |> Filename.basename |> basename_chop_extensions |> modulize
+(* We re-export the [Misc] definition, and ignore encoding errors under the
+   assumption that we should focus our effort on not *producing* badly encoded
+   module names *)
+let normalize x = match Misc.normalized_unit_filename x with
+  | Ok x | Error x -> x
+
+let stem source_file =
+  source_file |> Filename.basename |> basename_chop_extensions
 
-let start_char = function
-  | 'A' .. 'Z' -> true
-  | _ -> false
+let strict_modname_from_source source_file =
+  source_file |> stem |> strict_modulize
 
-let is_identchar_latin1 = function
-  | 'A'..'Z' | 'a'..'z' | '_' | '\192'..'\214' | '\216'..'\246'
-  | '\248'..'\255' | '\'' | '0'..'9' -> true
-  | _ -> false
+let modname_from_source source_file =
```

Comment:
What do you think of deprecating `modname_from_source` in favor of a new `unsafe_modname_from_source` ? It's currently used to construct idents in tools/ and in driver/ where strictness is not needed but the more verbose name couldn't hurt and might avoid bad uses outside of the compiler.

Reply:
I find `unsafe` a bit too much since the function is not `unsafe` by itself. I have settled for `lax_modname_from_source` to make the `strict` and `lax` variant more symmetric.